stdlib/public/core/Array.swift

```diff
@@ -1348,9 +1348,17 @@ extension Array: RangeReplaceableCollection {
     if !keepCapacity {
       _buffer = _Buffer()
     }
-    else {
+    else if _buffer.isMutableAndUniquelyReferenced() {
       self.replaceSubrange(indices, with: EmptyCollection())
     }
+    else {
+      let buffer = _ContiguousArrayBuffer<Element>(
+        _uninitializedCount: capacity,
+        minimumCapacity: 0
+      )
+      buffer.count = 0
```

Comment:
Minuscule nit: we can get rid of the assignment to `count` by swapping the init arguments. I think that would also make this read a little more straightforward, too:

```suggestion
      let buffer = _ContiguousArrayBuffer<Element>(
        _uninitializedCount: 0,
        minimumCapacity: capacity
      )
```

(With the original ordering, it seems weird that the desired count would be less than the capacity.)